clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java

```diff
@@ -73,20 +73,19 @@
 import io.grpc.Status;
 import io.grpc.StatusRuntimeException;
 import io.grpc.stub.StreamObserver;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.concurrent.BlockingDeque;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.LinkedBlockingDeque;
 import java.util.function.Supplier;
 
 public final class RecordingGatewayService extends GatewayImplBase {
 
-  private final BlockingDeque<GeneratedMessageV3> requests = new LinkedBlockingDeque<>();
```

Comment:
🔧 Those changes were introduced as a fix here: https://github.com/camunda/camunda/pull/19635
Should we rather backport the changes and keep them here as well?

Reply:
@mustafadagher I'm tagging you because you made those changes, what do you think ?

Reply:
The changes mainly reside in test, and they were aimed to fix flaky tests. it's safe to backport them if needed